pkg/segment/segment.go

```diff
@@ -0,0 +1,37 @@
+package segment
+
+import (
+	"github.com/segmentio/analytics-go/v3"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type Client struct {
+	client analytics.Client
+}
+
+var logger = logf.Log.WithName("segment_client")
+
+func NewClient(key string) (*Client, error) {
+	cl, err := analytics.NewWithConfig(key,
+		analytics.Config{
+			// Logger: nil,
```

Comment:
?

Reply:
ahem... forgot to implement an adapter to log when an event is sent to Segment 😂 
I will take care of it

Reply:
done in e701dec, there is a logger adapter which logs all info/errors. We can disable it later if it's too verbose.